import{an as t}from"./main-Cp9dG1PD.js";import{$ as o}from"./api-BBGbjhZB.js";const n=t({id:"record-store",state:()=>({errors:[],successMessage:""}),actions:{async fetchRecord(){try{return await o("/course-detail",{method:"GET",headers:{"Content-Type":"application/json"}})}catch(e){throw console.error("Error getting application list:",e),this.errors=e.response?e.response.data.errors:["An unexpected error occurred"],e}},async submitApplication(e){try{const r=await o("/course-detail",{method:"POST",body:e,headers:{"Content-Type":"application/json"}});return this.successMessage="Data submitted successfully",r.data}catch(r){throw console.error("Error submitting application:",r),this.errors=r.response?r.response.data.errors:["An unexpected error occurred"],r}},async submitEditApplication(e,r){try{const s=await o(`/course-detail/${e}`,{method:"PUT",body:r,headers:{"Content-Type":"application/json"}});return this.successMessage="Data submitted successfully",s.data}catch(s){throw console.error("Error submitting application:",s),this.errors=s.response?s.response.data.errors:["An unexpected error occurred"],s}},async deleteRecord(e){try{await o(`/course-detail/${e}`,{method:"DELETE"})}catch(r){console.error("Error deleting course details:",r),this.errors=r.response?r.response.data.errors:["An unexpected error occurred"]}}}});export{n as u};
